### PR TITLE
Set random seed

### PR DESCRIPTION
**Pull request by d61h6k4**
**Proposed changes**:
- ...

**Status (please check what you already did)**:
- [ ] added some tests for the functionality
- [ ] updated the documentation
- [x] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/main/changelog) for instructions)
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

File: rasa/utils/tensorflow/models.py [hunk 1]
```diff
@@ -97,9 +96,13 @@ def __init__(self, random_seed: Optional[int] = None, **kwargs: Any) -> None:
...
-        random.seed(self.random_seed)
-        tf.random.set_seed(self.random_seed)
-        np.random.seed(self.random_seed)
+        tf.keras.utils.set_random_seed(self.random_seed)
+        tf.experimental.numpy.random.seed(self.random_seed)
+        # When running on the CuDNN backend, two further options must be set
+        os.environ["TF_CUDNN_DETERMINISTIC"] = "1"
+        os.environ["TF_DETERMINISTIC_OPS"] = "1"
+        # Set a fixed value for the hash seed
+        os.environ["PYTHONHASHSEED"] = str(self.random_seed)
```
Comment (twerkmeister) [inline rasa/utils/tensorflow/models.py:105]: So the `random.seed` and `np.random.seed` is not necessary?

Comment (d61h6k4) [reply rasa/utils/tensorflow/models.py:105]: Right, because `tf.keras.utils.set_random_seed(self.random_seed)` inside calls something similar to:
```
import random
import numpy as np
import tensorflow as tf
random.seed(seed)
np.random.seed(seed)
tf.random.set_seed(seed)
```

Comment (d61h6k4) [reply rasa/utils/tensorflow/models.py]: I decided to make it explicit, and I returned them.

**Review by twerkmeister — APPROVED** [review]
